internal/apis/kfd/v1alpha2/ekscluster/common/preflight.go

```diff
@@ -50,6 +50,19 @@ type PreFlight struct {
 	InfrastructureTerraformOutputsPath string
 }
 
+func (p *PreFlight) getTerraformStateConfig() private.SpecToolsConfigurationTerraformStateS3 {
+	if p.FuryctlConf.Spec.ToolsConfiguration.Opentofu != nil {
+		return private.SpecToolsConfigurationTerraformStateS3{
```

Comment:
just curious: why do you need to return a new struct? can't you return here `p.FuryctlConf.Spec.ToolsConfiguration.Opentofu.State.S3` here like below for terraform?

Reply:
that makes sense, thanks!